src/error.rs

```diff
@@ -127,6 +131,26 @@ pub enum Error {
     /// Trailing data was found while parsing DER-encoded input for the named type.
     TrailingData(DerTypeId),
 
+    /// The certificate is not valid for the name it is being validated for.
+    ///
+    /// Variant without context for use in no-`alloc` environments. See `UnexpectedCertName` for
+    /// the variant that is used in `alloc` environments.
+    UnexpectedCertNameSimple,
```

Comment:
Opinions about the `Simple` suffix and the docstring? I figured it would be helpful to make the variant names similar but not the same.

Reply:
> /// Variant without context for use in no-`alloc` environments. See `UnexpectedCertName` for
> /// the variant that is used in `alloc` environments.

Did you consider taking the approach Rustls uses for [`OtherError`](https://github.com/rustls/rustls/blob/f98484bdbd57a57bafdd459db594e21c531f1b4a/rustls/src/error.rs#L639) where it's the same variant for both builds but the inner content differs?

It's an awkward construction but it also seems awkward to have a separate error variant to care about that will never happen if you always build w/ alloc. :thinking: 

> I figured it would be helpful to make the variant names similar but not the same.

I thought the original `CertNotValidForName` would be OK to leave as is. It's also similar to existing cert err variants like `CertNotValidYet`, and `CertExpired`. I would be minded to leave that one as-is or do a broader rework of the other variants. WDYT?